// model.proto
syntax = "proto3";
import "common.proto";

package trueno;

service Model {
  rpc ListModels (PingRequest) returns (ModelList) {}

  rpc CreateModel (ModelInfo) returns (ResultReply) {}
  rpc DistroConfig (ModelInfo) returns (ResultReply) {}
  rpc DeleteModel (ModelInfo) returns (ResultReply) {}

  rpc ExportModelImage (ModelInfo) returns (ResultReply) {}
  rpc ImportModelDistro (ModelInfo) returns (ResultReply) {}
}

/*
message ModelDistro {
  repeated float threshold = 1;
  repeated string mapping = 2;
  string hash = 3;
}
message ModelSpec {
  string version = 1;
  repeated ModelDistro dists = 2;
}
message ModelImpl {
  string head = 1;
  string bone = 2;
  string impl = 3;
  string hash = 4;
  repeated ModelSpec specs = 5;
}
message ModelInfo {
  string name = 1;
  repeated string labels = 2;
  string bundle = 3;
  repeated ModelImpl impls = 4;
}
*/

message ModelInfo {
  string name = 1;
  repeated string labels = 2;
  string bundle = 3;

  string head = 4;
  string bone = 5;
  string impl = 6;
  string implhash = 7;
  string version = 8;

  repeated string threshold = 9;
  repeated string mapping = 10;
  string modelext = 11;
  string disthash = 12;
}

message ModelList {
  repeated ModelInfo models = 1;
}

